doc/source/tutorial/part08/04_diels_alder.rst

```diff
@@ -0,0 +1,251 @@
+    This is the reaction site in the AbyU system. In the simulation we will fix
+    all atoms more than 20 Å from this site.
+
+Next, we need to recreate the system by adding the water sphere to protein and
+enzyme molecules:
+
+>>> new_mols = mols[:2] + water_sphere
+>>> system = sr.system.System()
+>>> for mol in new_mols:
+...     system.add(mol)
+
+.. note::
+
+    Here we add the molecules to the new system one-by-one to ensure that
+    the order is preserved.
+
+Now we'll add the original simulation box to the new system:
+
+>>> system.set_property("space", mols.property("space"))
+
+Next we'll minimise the system ready for simulation:
+
+>>> m = system.minimisation()
+>>> m.run()
+>>> mols = m.commit()
+
+Let's take a look at the system:
+
+.. image:: images/abyu.png
+   :target: images/abyu.png
+   :alt: AbyU system with a sphere of water molecules around the reaction site.
+
+Finally, we will write the system to an AMBER topology file for later use:
+
+>>> sr.save(mols, "abyu_sphere.prm7")
+
+Setting up the EMLE engine
+--------------------------
+
+We can now create an ``EMLECalculator`` to compute the QM intramolecular
+interaction and the electrostatic embedding interaction. Here we will use
+`xtb <https://xtb-docs.readthedocs.io/en/latest/>`_ as the QM backend:
+
+>>> from emle import EMLECalculator
+>>> calculator = EMLECalculator(backend="xtb", device="cpu")
+
+Next we will create an ``EMLEEngine`` that can be be used to perform QM/MM
+calculation:
+
+>>> qm_mols, engine = sr.qm.emle(
+...     mols,
+...     "atomnum 1804:1822,2083:2132",
+...     calculator,
+...     redistribute_charge=True
+... )
+
+Here the selection for the QM region includes tryptophan side-chain atoms
+(1804-1822) and the substrate (2083-2132). The ``redistribute_charge`` keyword
+ensures that the charge on atoms in the QM region is integer valued by
+redistributing the remaining fractional charge over MM atoms within the
+residues containing the QM atoms.
+
+Creating a context
+------------------
+
+We can now create a dynamics object that will create an ``OpenMM`` context for
+us.  In order to use the solvent sphere we will need to specify the ``fixed``
+keyword argument. This specifies a selection for the atoms that should be
+fixed during simulation. Here we will fix all atoms more than 20 Å from the
+reaction site:
+
+>>> d = mols.dynamics(
+...     timestep="1fs",
+...     constraint="none",
+...     perturbable_constraint="none",
+...     integrator="langevin_middle",
+...     cutoff_type="rf",
+...     qm_engine=engine,
+...     platform="cpu",
+...     fixed="not atoms within 20A of atomidx 3 in molidx 1",
+)
+
+.. note::
+
+    In ``OpenMM``, fixed atoms are implemented by setting atomic masses to zero.
+    This means that the atoms are still involved in interactions, but do not move.
+
+Now we will extract the context from the dynamics object:
+
+>>> context = d._d._omm_mols
```

Comment:
Would it be worth adding a `d.extract_context()` function (or similar) that would make getting the context a more "official" operation? The tutorial shows that this is a very useful thing to do. It is a bit messy to do this by calling private functions.

Reply:
That's a good idea. I'll add this in as it will make the tutorial a lot cleaner.

Reply:
I've gone for `d.context()` since there's an equivalent `d.integrator()` already that gets the OpenMM integrator.